### DISCUSSION TITLE
Long context CLIP

### DISCUSSION THREAD

**Discussion by nicolas-dufour** [mlfoundations/open_clip#876] [Q&A]
Hi,
Is there any CLIP model that manages longer context length than 77 (ideally >256)

Is there a reason why the context length is set to 77? LAION has too short alt-texts overall?

Thanks!

**Comment by rom1504** [comment]
1. You need data that has meaningful long captions 
2. You need eval tasks where meaningful long captions are important (so not zero shot classification and not most caption retrieval evals)

**Comment by BIGBALLON** [comment]
@nicolas-dufour @rom1504 

BTW： maybe [Long-CLIP](https://github.com/beichenzbc/Long-CLIP) is your need? 

- 🔥 Long Input length Increase the maximum input length of CLIP from **77 to 248.**
- 🔥 Strong performance Improve the R@5 of long-caption text-image retrieval by 20% and traditional text-image retrieval by 6%.
- 🔥 Plug-in and play Can be directly applied in any work that requires long-text capability.

**Comment by rwightman** [comment]
Moving this to discussions now for reference.

**Comment by PixelChen24** [comment]
I wonder whether replacing the default tokenizer with others will work?

**Comment by sachinruk** [comment]
Just wanted to double check. Does this mean that **all** models in this library have a context length of 77 at most?

**Reply by rwightman** [reply]
@sachinruk yes, they were trained from scratch with noisy internet image-text web data (openai wit, laion, datacomp, dfn, webli) that typically has fairly short captions so 32-77 tokens is the range here. 

Having quality text beyond that range requires either adapting an existing longer context LLM as part of a VLM, or if training from scratch, a LOT of image-text data with higher quality, longer captions (which would be a challenge an billion scale). 

This one might be the one exception https://huggingface.co/laion/CLIP-ViT-B-32-xlm-roberta-base-laion5B-s13B-b90k .. it was an experiment using an existing text encoder.

Could also fine-tune one of these existing models on a decent size image-text data with longer captions and increase the context length...

**Comment by rom1504** [comment]
I think the first thing to solve on this topic is producing a good long
captions image text *open* large scale dataset. Is there anything yet?

If not, probably it would mean running one of the small VLM on 1B images

**Comment by rom1504** [comment]
The second very important thing is having tasks to evaluate the meaning of
long text alignment with images.

I am guessing the interest here is about using CLIP as an evaluator or
conditioner for GenAI.

In that case you may want to consider whether you really need full
attention understanding of the text and the link of it with images.

If you actually don't... Then what about cutting your text in N pieces of
size 77 and pooling the resulting embeddings? You can get infinite context
length by doing that.
